database/database.go

```diff
@@ -0,0 +1,255 @@
+	SkynetDBUserEV = "SKYNET_DB_USER"
+	// SkynetDBPassEV holds the name of the environment variable for DB password.
+	SkynetDBPassEV = "SKYNET_DB_PASS"
+
+	// SkynetDBName defines the name of Skynet's database.
+	SkynetDBName = "skynet"
+	// SkynetDBUsersCollection defines the name of the "users" collection within
+	// skynet's database.
+	SkynetDBUsersCollection = "users"
+
+	// ErrUserNotFound is returned when we can't find the user in question.
+	ErrUserNotFound = errors.New("user not found")
+	// ErrEmailAlreadyUsed is returned when we try to use an email to either
+	// either create or update a user and another user already uses this email.
+	ErrEmailAlreadyUsed = errors.New("email already in use by another user")
+	// ErrNoDBConnection is returned when we can't connect to the database.
+	ErrNoDBConnection = errors.New("no connection to the database")
+
+	// True is a helper, so we can easily provide a *bool for UpdateOptions.
+	True = true
+	// False is a helper, so we can easily provide a *bool for UpdateOptions.
+	False = false
+)
+
+// DB represents a MongoDB database connection.
+type DB struct {
+	ctx   context.Context
+	db    *mongo.Database
+	users *mongo.Collection
+}
+
+// New returns a new DB connection based on the environment variables.
+func New(ctx context.Context) (*DB, error) {
+	opts, err := connectionOptionsFromEnv()
+	if err != nil {
+		return nil, errors.AddContext(err, "failed to get all necessary connection parameters")
+	}
+	return NewCustom(ctx, opts[SkynetDBUserEV], opts[SkynetDBPassEV], opts[SkynetDBHostEV], opts[SkynetDBPortEV], SkynetDBName)
+}
+
+// NewCustom returns a new DB connection based on the passed parameters.
+func NewCustom(ctx context.Context, user, pass, host, port, dbname string) (*DB, error) {
+	connStr := fmt.Sprintf("mongodb://%s:%s@%s:%s/?compressors=zlib&gssapiServiceName=mongodb&readPreference=nearest", user, pass, host, port)
+	c, err := mongo.NewClient(options.Client().ApplyURI(connStr))
+	if err != nil {
+		return nil, errors.AddContext(err, "failed to create a new DB client")
+	}
+	err = c.Connect(ctx)
+	if err != nil {
+		return nil, errors.AddContext(err, "failed to connect to DB")
+	}
+	database := c.Database(dbname)
+	users := database.Collection(SkynetDBUsersCollection)
+	db := &DB{
+		ctx:   ctx,
+		db:    database,
+		users: users,
+	}
+	return db, nil
+}
+
+// Context returns the DB's internal context, so we can build upon it.
+func (db *DB) Context() context.Context {
+	return db.ctx
+}
+
+// Disconnect closes the connection to the database in an orderly fashion.
+func (db *DB) Disconnect(ctx context.Context) error {
+	if db.db == nil {
+		return ErrNoDBConnection
+	}
+	return db.db.Client().Disconnect(ctx)
+}
+
+// UserDeleteByID deletes a user by their ID and returns `false` in case there
+// was an error or the user was not found.
+func (db *DB) UserDeleteByID(ctx context.Context, id string) (bool, error) {
```

Comment:
I don't think we are ever going to expose the ID through the API right? So we can probably provide a `User` as input here since we need to fetch it first anyway.

Reply:
Right, I was thinking about that. Yes, it makes sense. And if the user doesn't have an ID in it yet then we'll just error out.